cmd/kops-controller/controllers/node_controller.go

```diff
@@ -104,26 +85,12 @@ func (r *NodeReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	cluster, err := r.getClusterForNode(node)
+	info, err := r.identifier.IdentifyNode(ctx, node)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("unable to load cluster object for node %s: %v", node.Name, err)
+		return ctrl.Result{}, fmt.Errorf("error identifying node %q: %v", node.Name, err)
 	}
 
-	ig, err := r.getInstanceGroupForNode(ctx, node)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("unable to load instance group object for node %s: %v", node.Name, err)
-	}
-
-	labels := nodelabels.BuildNodeLabels(cluster, ig)
-
-	lifecycle, err := r.getInstanceLifecycle(ctx, node)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("unable to get instance lifecycle %s: %v", node.Name, err)
-	}
-
-	if len(lifecycle) > 0 {
-		labels[fmt.Sprintf("node-role.kubernetes.io/%s-worker", lifecycle)] = "true"
```

Comment:
Did we decide what to do about this node-role?  (It looks like we're no longer going to set it?)

Reply:
I moved this logic into the cloud-provider-specific code. It just includes this label in the list of labels it returns.